addons/mail/models/discuss/discuss_channel.py

```diff
@@ -988,6 +985,22 @@ def _set_last_seen_message(self, last_message, allow_older=False):
             'seen_message_id': last_message.id,
             'last_seen_dt': fields.Datetime.now(),
         })
+        data = {
+            'channel_id': self.id,
+            'last_message_id': last_message.id,
+            'member_id': member.id,
+        }
+        current_partner, current_guest = self.env["res.partner"]._get_current_persona()
+        data['partner_id' if current_partner else 'guest_id'] = current_partner.id if current_partner else current_guest.id
+        target = current_partner or current_guest
+        if self.channel_type in self._types_allowing_seen_infos_on_self():
+            target = self
+        self.env['bus.bus']._sendone(target, 'discuss.channel.member/seen', data)
+
+    def _types_allowing_seen_infos_on_self(self):
```

Comment:
self is confusing in this context because it's not clear whether it refers to the user or the channel